compiler/fir/resolve/src/org/jetbrains/kotlin/fir/types/TypeUtils.kt

```diff
@@ -204,10 +204,17 @@ fun FirTypeRef.withReplacedReturnType(newType: ConeKotlinType?): FirTypeRef {
     require(this is FirResolvedTypeRef || newType == null)
     if (newType == null) return this
 
-    return buildResolvedTypeRef {
-        source = this@withReplacedReturnType.source
-        type = newType
-        annotations += this@withReplacedReturnType.annotations
+    return if (newType is ConeKotlinErrorType) {
```

Comment:
Why don't you want to call `resolvedTypeFromPrototype` also here?

Reply:
Ah, didn't think about that!